x/stakeibc/keeper/msg_server.go

```diff
@@ -277,6 +285,23 @@ func (k msgServer) RegisterHostZone(goCtx context.Context, msg *types.MsgRegiste
 	return &types.MsgRegisterHostZoneResponse{}, nil
 }
 
+func (ms msgServer) UpdateHostZoneParams(goCtx context.Context, msg *types.MsgUpdateHostZoneParams) (*types.MsgUpdateHostZoneParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	if ms.authority != msg.Authority {
+		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", ms.authority, msg.Authority)
+	}
+
+	hostZone, found := ms.Keeper.GetHostZone(ctx, msg.ChainId)
+	if !found {
+		return nil, types.ErrHostZoneNotFound.Wrapf("host zone %s not found", msg.ChainId)
+	}
+
+	hostZone.MaxMessagesPerIcaTx = msg.MaxMessagesPerIcaTx
```

Comment:
Should we do any validation on `msg.MaxMessagesPerIcaTx`? Possible it's handled elsewhere already!

Reply:
good call - forgot the default coalesce was in the main register function. Just updated!

Reply:
Nice, resolving!